jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextRequest.java

```diff
@@ -190,6 +190,11 @@ public HttpURI getHttpURI()
             decodedPathInContext,
             matchedResource
         );
+        if (request instanceof ServletContextRequest originalServletContextRequest)
```

Comment:
But this will not work if the request is further wrapped.   So you probably need something like:
```suggestion
        ServletContextRequest originalServletContextRequest = Request.as(request, ServletContextRequest.class);
        if (originalServletContextRequest != null)
```

The `Request.as` will unwrap wrappers looking for the type.... but now I'm worried that will see through a cross context dispatch.  Perhaps we need to modify as so that it does not see through cross context dispatch?

Reply:
@gregw ok, so how about we check that a context boundary isn't crossed as we do the unwrap?  See commit https://github.com/jetty/jetty.project/pull/13041/commits/62852e56321260f65b8cd5d22a30148b3f222e23